src/app/plugins/leaderboard.plugin.ts

```diff
@@ -0,0 +1,133 @@
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IMessage, Maybe } from '../../common/types';
+import { GameType } from '../../services/gameleaderboard.service';
+
+export class LeaderboardPlugin extends Plugin {
+  public name: string = 'Leaderboard Plugin';
+  public description: string = 'Gets the leaderboards of games';
+  public usage: string = 'leaderboard <game (optional)>';
+  public pluginAlias = ['lb'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName = Constants.Channels.Public.Games;
+
+  private readonly _MENTION_REGEX = /<@!?&?(\d+)>/;
+
+  public validate(message: IMessage, args: string[]) {
+    return args.length >= 1;
+  }
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const [gameName, opponentOne, opponentTwo] = args;
+    const gameEnum: Maybe<GameType> = this._convertGameNameToEnum(gameName);
+    if (!gameEnum) {
+      message.reply(`Couldn't find that game`);
+      return;
+    }
+
+    const guild = message.guild;
+    if (!guild) {
+      message.reply('Please use this command in a guild');
+      return;
+    }
+
+    // Get default leaderboard if no users are given
+    if (!opponentOne) {
+      const embed = await this.container.gameLeaderboardService.createOverallLeaderboardEmbed(
+        message.author,
+        gameEnum
+      );
+
+      message.channel.send(embed);
+      return;
+    }
+
+    // Give one players leaderboard if no opponent is given
+    if (!opponentTwo) {
+      const embed = await this._createOpponentPlayerEmbed(message, opponentOne, guild, gameEnum);
+      await message.channel.send(embed || 'Error getting leaderboards');
+      return;
+    }
+
+    const embed = await this._getMatchUpEmbed(message, opponentOne, opponentTwo, guild, gameEnum);
+    await message.channel.send(embed || 'Error getting leaderboards');
+  }
+
+  private async _createOpponentPlayerEmbed(
+    message: IMessage,
+    opponent: string,
+    guild: Guild,
+    gameEnum: GameType
+  ) {
+    const match = opponent.match(this._MENTION_REGEX);
+    if (!match) {
+      message.reply('Invalid <matchup> argument');
+      return null;
+    }
+
+    // The ID is is the first group of the match
+    const [, uID] = match;
+    const oppUser = guild.members.cache.get(uID)?.user;
+
+    // user could not be found
+    if (!oppUser) {
+      message.channel.send('User could not be found');
+      return null;
+    }
+
+    return this.container.gameLeaderboardService.createMatchupLeaderboardEmbed(
+      message.author,
+      oppUser,
+      gameEnum
+    );
+  }
+
+  private _getMatchUpEmbed(
+    message: IMessage,
+    playerOneString: string,
+    playerTwoString: string,
+    guild: Guild,
+    gameEnum: GameType
+  ) {
+    const matchOne = playerOneString.match(this._MENTION_REGEX);
+    const matchTwo = playerTwoString.match(this._MENTION_REGEX);
```

Comment:
live ur best life but i would write it as:
`const [matchOne, matchTwo] = [playerOneString, playerTwoString].map(p => p.match(this._MENTION_REGEX))`

Reply:
agree